backend/src/services/wallet_service.py

```
@@ -121,8 +145,34 @@ async def get_wallet_balance(self) -> Dict:
+            total_balance_raw = 0
+            for account_info in response.value:
+                 data = account_info.account.data
+                 amount_raw = int.from_bytes(data[64:72], "little")
+                 total_balance_raw += amount_raw
+
+            if total_balance_raw == 0:
+                return 0.0
+
+            decimals = await self._get_token_decimals(mint_address)
+            return total_balance_raw / (10 ** decimals)
+        except Exception as e:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)